src/validation.cpp

```diff
@@ -5252,6 +5252,14 @@ bool ChainstateManager::ActivateSnapshot(
         return false;
     }
 
+    auto maybe_au_data = ExpectedAssumeutxo(BlockHash{base_blockhash}, ::Params());
```

Comment:
huh? This absolutely avoids a crash. Do you have steps to not-reproduce the crash I added in that commit?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/21584#discussion_r612257797

> huh? This absolutely avoids a crash. Do you have steps to not-reproduce the crash I added in that commit?

Marco. You're going to make me actually test a PR? What about my tried and true method of just asking stupid questions about PRs and hoping that it sometimes makes things better?

But just in case anyone else is curious where the crash happens and doesn't want to endure the indignities Marco is trying to subject me to, the crash doesn't happen where I was looking for it in `ActivateSnapshot`, but actually in an assert inside the `coins_cache.Flush()` call in `PopulateAndValidateSnapshot`:

https://github.com/bitcoin/bitcoin/blob/a1f0b8b62eb851c837a3618583b7c2fd4d12006c/src/validation.cpp#L5275

The stack trace is:

````
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
#1  0x00007ffff583c8b1 in __GI_abort () at abort.c:79
#2  0x00007ffff582c42a in __assert_fail_base (fmt=0x7ffff59b3a38 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n", 
    assertion=assertion@entry=0x555556c8b450 "!hashBlock.IsNull()", file=file@entry=0x555556c8b464 "txdb.cpp", line=line@entry=89, 
    function=function@entry=0x555556c8b46d "virtual bool CCoinsViewDB::BatchWrite(CCoinsMap &, const uint256 &)") at assert.c:92
#3  0x00007ffff582c4a2 in __GI___assert_fail (assertion=0x555556c8b450 "!hashBlock.IsNull()", file=0x555556c8b464 "txdb.cpp", line=89, 
    function=0x555556c8b46d "virtual bool CCoinsViewDB::BatchWrite(CCoinsMap &, const uint256 &)") at assert.c:101
#4  0x000055555636738b in CCoinsViewDB::BatchWrite (this=0x5555577975c0, mapCoins=std::unordered_map with 110 elements = {...}, hashBlock=...) at txdb.cpp:89
#5  0x00005555564a2e80 in CCoinsViewBacked::BatchWrite (this=0x5555577975f8, mapCoins=std::unordered_map with 110 elements = {...}, hashBlock=...) at coins.cpp:30
#6  0x00005555564a43de in CCoinsViewCache::Flush (this=0x55555778eaf0) at coins.cpp:223
#7  0x00005555563fc11d in ChainstateManager::PopulateAndValidateSnapshot (this=0x55555740b038 <g_chainman>, snapshot_chainstate=..., coins_file=..., metadata=...)
    at validation.cpp:5422
#8  0x00005555563fab3d in ChainstateManager::ActivateSnapshot (this=0x55555740b038 <g_chainman>, coins_file=..., metadata=..., in_memory=true) at validation.cpp:5299
#9  0x0000555555e8c893 in validation_chainstatemanager_tests::CreateAndActivateUTXOSnapshot<validation_chainstatemanager_tests::chainstatemanager_activate_snapshot::test_method()::$_12>(NodeContext&, boost::filesystem::path, validation_chainstatemanager_tests::chainstatemanager_activate_snapshot::test_method()::$_12) (node=..., 
    root=..., malleation=...) at test/validation_chainstatemanager_tests.cpp:199
#10 0x0000555555e8877a in validation_chainstatemanager_tests::chainstatemanager_activate_snapshot::test_method (this=0x7fffffffc8d0)
    at test/validation_chainstatemanager_tests.cpp:262
```

Reply:
Thanks Ryan, added your stack trace to https://github.com/bitcoin/bitcoin/pull/21584#issue-608396741

Reply:
> You're going to make me actually test a PR

The commit that added the assert is mentioned in OP, so in theory it would be possible to review without testing ;)

Reply:
> The commit that added the assert is mentioned in OP, so in theory it would be possible to review without testing ;)

Now I get it. Thanks! Another connection I didn't make the first time around.